src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -834,23 +770,24 @@ int D4NFilterObject::D4NFilterReadOp::D4NFilterGetCB::handle_data(bufferlist& bl
 
       bl.splice(0, len_to_copy, &bl_copy);
       bl_rem.claim_append(bl_copy);
-      block.hostsList.push_back(blockDir->get_addr().host + ":" + std::to_string(blockDir->get_addr().port));
-      block.size = source->get_obj_size();
-      block.cacheObj.bucketName = source->get_bucket()->get_name();
+      block.size = source->get_obj_size(); // TODO: fill out block entirely
+      block.hostsList.push_back("127.0.0.1:6379" /*current cache addr*/); 
```

Comment:
Two general comments here:
1. The 'insert' method has to be moved to the cache driver
2. The block size is incorrect, please refer to the PR in which I have added these values.